models/common.go

```diff
@@ -148,3 +151,119 @@ type TaggableModel interface {
 type LabelableModel interface {
 	GetLabels() map[string]string
 }
+
+type ColumnType string
+
+const (
+	ColumnTypeBoolean  ColumnType = "boolean"
+	ColumnTypeDateTime ColumnType = "datetime"
+	ColumnTypeDecimal  ColumnType = "decimal"
+	ColumnTypeDuration ColumnType = "duration"
+	ColumnTypeInteger  ColumnType = "integer"
+	ColumnTypeJSONB    ColumnType = "jsonb"
+	ColumnTypeString   ColumnType = "string"
+)
+
+// ConvertViewRecordsToNativeTypes converts view cell to native go types
+func ConvertViewRecordsToNativeTypes(row map[string]any, columnDef map[string]ColumnType) (map[string]any, map[string][]string) {
+	// keep track of all the invalid types encountered per column
+	invalidTypesPerColumn := make(map[string][]string)
```

Comment:
We don't want an error per cell, only per row, if their is a casting issue it is likely to be an issue for all rows